### PR TITLE
feat(FixedFooterLayout): stop using Portal for the fixed part

### PR DESCRIPTION
**Pull request by pladaria**
WEB-339

Related PR: https://github.com/Telefonica/webapp/pull/904

* [x] Verify in multiple webapp flows
* [x] Verify in mobile browsers (iOS + Android)
* [x] Verify in movistar.es website (embedded features)
* [x] Verify that [this iOS bug](https://jira.tid.es/browse/APPS-7485) is not reproduced. 
    * Verified in iOS 13 and 14

### CODE REVIEW

File: src/fixed-footer-layout.tsx [hunk 1]
```diff
@@ -4,26 +4,18 @@ import debounce from 'lodash/debounce';
...
 
 const getScrollDistanceToBottom = () => document.body.scrollHeight - window.innerHeight - window.scrollY;
 
-const waitForSwitchTransitionToStart = (fn: () => void) => {
-    const timeoutId = setTimeout(fn, 0);
-    return {
-        cancel: () => clearTimeout(timeoutId),
-    };
-};
-
```
Comment (pladaria) [inline src/fixed-footer-layout.tsx:18]: I don't remember why was this needed

File: src/fixed-footer-layout.tsx [hunk 2]
```diff
@@ -35,6 +34,7 @@ const useStyles = createUseStyles((theme) => ({
     containerSmall: {
         paddingBottom: ({height}) => height,
         backgroundColor: ({containerBgColor}) => containerBgColor || theme.colors.background,
+        minHeight: ({topDistance}) => `calc(100vh - ${topDistance}px)`,
```
Comment (anabelengp) [inline src/fixed-footer-layout.tsx:37]: I think that with this change, I can get rid of this div  I had to mainain in order to fill the background color in feedbackScreen.
https://github.com/Telefonica/mistica-web/pull/343/commits/782f7c84726f0947b2e0c4fe5b4badd83dca13d9#diff-a73f3b991d7fd52028c47266987d26b2caef049af23f23ec6adffd8c12d9231aR276
We have to remember to check it after mergin any of the two PRs.

**Review by anabelengp — APPROVED** [review]

File: src/fixed-footer-layout.tsx (hunk 2 above)
Comment (pladaria) [reply src/fixed-footer-layout.tsx:37]: This could be not correct :/ At least not with some usages I'm seeing in webapp...

**Review by atabel — APPROVED** [review]

**Review by Marcosld — APPROVED** [review]

File: src/fixed-footer-layout.tsx [hunk 3]
```diff
@@ -105,44 +100,49 @@ const FixedFooterLayout: React.FC<Props> = ({
         );
 
         const transitionAwaiter = waitForSwitchTransitionToStart(checkDisplayShadow);
-        addPassiveEventListener(window, 'resize', checkDisplayShadow);
-        addPassiveEventListener(window, 'scroll', checkDisplayShadow);
+        const scrollEventTarget = getScrollEventTarget(scrollable);
+        addPassiveEventListener(scrollEventTarget, 'resize', checkDisplayShadow);
```
Comment (atabel) [inline src/fixed-footer-layout.tsx:104]: resize event listener should be on window, you cannot listen to dom elements resize without a ResizeObserver

**Review by atabel — APPROVED** [review]

File: src/fixed-footer-layout.tsx [hunk 4]
```diff
@@ -4,10 +4,13 @@ import debounce from 'lodash/debounce';
...
-import {Portal} from './portal';
-import {addPassiveEventListener, removePassiveEventListener} from './utils/dom';
+import {addPassiveEventListener, getScrollableParentElement, removePassiveEventListener} from './utils/dom';
 
-const getScrollDistanceToBottom = () => document.body.scrollHeight - window.innerHeight - window.scrollY;
+const getScrollDistanceToBottom = (el: HTMLElement) => el.scrollHeight - el.scrollTop - el.clientHeight;
+
+const getScrollEventTarget = (el: HTMLElement) => (el === document.documentElement ? window : el);
+
+const hasScroll = (el: HTMLElement) => el.scrollHeight > el.clientHeight;
```
Comment (anabelengp) [inline src/fixed-footer-layout.tsx:13]: could these functions be utility functions in dom.tsx?

**Review by anabelengp — APPROVED** [review]

File: src/fixed-footer-layout.tsx (hunk 2 above)
Comment (anabelengp) [reply src/fixed-footer-layout.tsx]: I will keep the div then

File: src/fixed-footer-layout.tsx (hunk 4 above)
Comment (pladaria) [reply src/fixed-footer-layout.tsx]: sure! moved `getScrollDistanceToBottom` and `hasScroll`

File: src/fixed-footer-layout.tsx (hunk 3 above)
Comment (pladaria) [reply src/fixed-footer-layout.tsx:106]: not an issue for now. I'm not aware of a fixed footer usage where the scrollable parent is not the html element. In that case the scrollEventTarget is window.
